File: InJava/app/src/main/java/com/barryzeha/quizquestion/view/RankingActivity.java
package com.barryzeha.quizquestion.view;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import com.barryzeha.quizquestion.R;
import com.barryzeha.quizquestion.databinding.ActivityRankingBinding;
import com.barryzeha.quizquestion.model.adapters.RankingAdapter;
import com.barryzeha.quizquestion.model.common.Util;
import com.barryzeha.quizquestion.viewModel.RankingViewModel;

import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class RankingActivity extends AppCompatActivity {
    private  ActivityRankingBinding bind;
    private RankingViewModel rankingViewModel;
    private RankingAdapter rankingAdapter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        bind= ActivityRankingBinding.inflate(getLayoutInflater());
        Util.loadUrl(R.drawable.ranking_background,bind.ivBackgroundRanking,this);
        setContentView(bind.getRoot());
        
        setUpAdapter();
        setUpViewModel();
        bind.btnHome.setOnClickListener(v->{
            Intent intent = new Intent(this, MainActivity.class);
            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
            startActivity(intent);
        });
    }

    private void setUpViewModel() {
        rankingViewModel= new ViewModelProvider(this).get(RankingViewModel.class);
        new Thread(() -> {
                rankingViewModel.callRanking();
       }).start();

        rankingViewModel.getRanking().observe(this,listRanking->{
            rankingAdapter.addRanking(listRanking);
        });
    }

    private void setUpAdapter() {
        rankingAdapter= new RankingAdapter();
        bind.rvRanking.setHasFixedSize(true);
        bind.rvRanking.setLayoutManager(new LinearLayoutManager(this));
        bind.rvRanking.setAdapter(rankingAdapter);
    }
}

File: InKotlin/app/src/main/java/com/barryzeha/quizquestion/view/RankingActivity.kt
package com.barryzeha.quizquestion.view

import android.content.Intent
import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.barryzeha.quizquestion.R
import com.barryzeha.quizquestion.common.loadUrl
import com.barryzeha.quizquestion.common.startActivity
import com.barryzeha.quizquestion.databinding.ActivityRankingBinding
import com.barryzeha.quizquestion.model.adapters.RankingAdapter
import com.barryzeha.quizquestion.viewModel.RankingViewModel
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

@AndroidEntryPoint
class RankingActivity : AppCompatActivity() {
    private lateinit var bind : ActivityRankingBinding
    private val rankingViewModel:RankingViewModel by viewModels()
    private var adapter:RankingAdapter?=null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        bind = ActivityRankingBinding.inflate(layoutInflater)
        setContentView(bind.root)
        bind.ivBackgroundRanking.loadUrl(R.drawable.ranking_background)
        setUpViewModel()
        setUpAdapter()
        bind.btnHome.setOnClickListener { startActivity<MainActivity> {
            flags= Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_CLEAR_TASK
        } }

    }
    private fun setUpViewModel() {
        CoroutineScope(Dispatchers.Main).launch {
            rankingViewModel.getRanking().observe(this@RankingActivity) { rankingList ->
                adapter?.addRanking(rankingList)

            }
        }

    }
    private fun setUpAdapter()= with(bind){

        adapter= RankingAdapter()
        rvRanking.setHasFixedSize(true)
        rvRanking.layoutManager=LinearLayoutManager(this@RankingActivity)
        rvRanking.adapter=adapter
    }
}